src/screens/OMPSwitch/OMPSwitchHelper.res

```diff
@@ -35,20 +35,14 @@ module AddNewMerchantProfileButton = {
     ~customHRTagStyle="",
     ~addItemBtnStyle="",
   ) => {
-    let {userHasAccess, hasAnyGroupAccess} = GroupACLHooks.useUserGroupACLHook()
-    let cursorStyles = GroupAccessUtils.cursorStyles(
-      // TODO: Remove `MerchantDetailsManage` permission in future
-      hasAnyGroupAccess(
-        userHasAccess(~groupAccess=MerchantDetailsManage),
-        userHasAccess(~groupAccess=AccountManage),
-      ),
-    )
+    open OMPSwitchUtils
+    let {userInfo: {roleId}} = React.useContext(UserInfoProvider.defaultContext)
+
+    let cursorStyles = GroupAccessUtils.cursorStyles(hasCreateNewOMPAccess(user, roleId))
+
     <ACLDiv
-      // TODO: Remove `MerchantDetailsManage` permission in future
-      authorization={hasAnyGroupAccess(
-        userHasAccess(~groupAccess=MerchantDetailsManage),
-        userHasAccess(~groupAccess=AccountManage),
-      )}
+      authorization={hasCreateNewOMPAccess(user, roleId)}
+      noAccessDescription="You do not have the required permissions for this action. Please contact your admin."
```

Comment:
There will be no need for noAccessDescription. This will be automatically controlled by noAccessControlText and authorization .

Reply:
Yeah `noAccessControlText` has a description but that is not aligned with what we are showing over here, the buttons require custom `noAccessControlText `